src/controllers/events.js

```diff
@@ -0,0 +1,39 @@
+const Event = require('../models/event')
+const { uploadImage } = require('../db/storage')
+const { getFileExtension } = require('../utility/utils');
+const {EVENT_IMAGE_DIR} = require('../utility/variables')
+
+const createEvent = async (req, res) => {
+    try {
+        const imageUrl = await uploadImage(
+            req.file,
+            `${EVENT_IMAGE_DIR}/${Date.now()}.${getFileExtension(
+                req.file.originalname
+            )}`
+        );
+        const event = await new Event({
```

Comment:
Now, this event does not have an id. Instead you can say something like: 
```
const event = await Event.create({...})  
```
which will return an event with an id that you can use while uploading the image. 
After that to save the URL of the image you can say: 
```
event.coverImage = imageUrl; 
await event.save();
```